### DISCUSSION TITLE
Unable to configure Nuxt 3 with vite & hmr to be run with Gitpod

### DISCUSSION THREAD

**Discussion by davemecha** [nuxt/nuxt#16293] [Questions]
### Environment

Nuxi 3.0.0-rc.11                                                                                              17:33:35
RootDir: /workspace/nuxt-gitpod-test                                                                          17:33:35
Nuxt project info:                                                                                            17:33:35

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.17.1`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `buildModules`, `typescript`, `vite`, `build`, `css`
- Runtime Modules:  `-`
- Build Modules:    `nuxt-vite@0.3.5`, `@nuxtjs/tailwindcss@5.3.3`, `@pinia/nuxt@0.4.2`
------------------------------

### Reproduction

I created a demo repository https://github.com/davemecha/nuxt-gitpod-test

* Just start a VSCode Gitpod environment in the browser for this repo with: https://gitpod.io/#https://github.com/davemecha/nuxt-gitpod-test
* Make sure the dev server is started in the terminal (should start automatically, but the data collection confirmation must be passed so finish startup). it can be started manually with `start dev`
* Open ports view (in the bottom, next to the terminal)
* Open port 3000 by clicking the globe icon
* Page is in reload cycle since vite dev server is not configured correctly (check with chrome dev tools)

### Describe the bug

I can't find any way to configure HMR with vite so it runs as expected behind the reverse proxy of Gitpod. The websocket connection cannot be established correctly.

I would expect something similar to `clientPort` for a client URL but was not able to find a proper setting for this. I also tried to somehow use a proxy config in the nuxt config to work around the issue but I was out of luck here too.

### Additional context

I'm new to Nuxt, so I think I missed something. It would be great to find a solution to this. I would add this solution to the demo repo posted above, so future users of remote coding environments like Gitpod find it easier to work with Nuxt and vite.

I also opened a [Stackoverflow](https://stackoverflow.com/questions/73999822/how-to-cofigure-hmr-with-nuxt-3-vite-on-gitpod-web-behind-proxy) question on that matter and discussed it with the Gitpod community on their [Discord](https://discord.com/channels/816244985187008514/1028386778052313170). Unfortunately nobody was able to help yet.

**Comment by davemecha** [comment]
I played a bit more with the proxy config of vite. I was thinking, maybe I can define some special route for the hmr server.

With the `hmr` part in the config the hmr request is sent to `wss://3000-WORKSPACE_ID.gitpod.io/_nuxt/myhmr` instead of `ws://localhost:1337`. But unfortunately the hmr server is listening on `ws://localhost:1337` or `wss://1337-WORKSPACE_ID.gitpod.io/`.

So I tried to set a proxy in the vite config. In general a proxy endpoint works as expected. Only when it starts with `_nuxt` it's always redirected to the main page on 3000.

``` js
export default defineNuxtConfig({

  ...

  vite: {
    server: {
      hmr: {
        port: 1337,
        protocol: 'wss',
        clientPort: 443,
        path: 'myhmr',
      },
      proxy: {
        '/_nuxt/myhmr': {
          target: 'http://localhost:1337',
          // changeOrigin: true,
          // secure: false,
          ws: true,
        },
      },
    },
  },

  ...

});
```

So I'm still stuck... :(

I'd be glad for any help....

**Reply by wvffle** [reply]
I've tried a lot and I am still stuck with no luck at all..

**Reply by davemecha** [reply]
Yeah, same with me. I don't get it running. My workaround is using Gitpod with a local IDE - But that's not really helpful in many situations. On-boarding of new devs takes longer, multiple instances of a project are not possible due to local port conflicts, etc... - 

Too bad, supporting cloud coding environments doesn't seem too relevant for the nuxt team for now.
